packages/logger/src/lib/logger.ts

```diff
@@ -427,14 +435,20 @@ export class LogManager {
   }
 
   get LoggerIds(): string[] {
-    const keys: string[] = [];
+    const keys: [string, number][] = [];
     for (const category of this._loggers.entries()) {
       for (const child of category[1].Children) {
-        keys.push(child[0]);
+        keys.push([child[0], child[1].timestamp]);
       }
     }
 
-    return keys;
+    return keys
+      .sort((a: [string, number], b: [string, number]) => {
```

Comment:
the goal here is to sort the array keys based on the timestamp (child[1].timestamp]) right? If so, I think we need to do the following instead, because the sorting callback should return a positive number, a negative number, and a 0 if they are equal.

```js
  return keys
    .sort((a: [string, number], b: [string, number]) => {
        return a[1] - b[1];
    }) // Corrected sorting
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

> A comparator conforming to the constraints above will always be able to return all of 1, 0, and -1, or consistently return 0. For example, if a comparator only returns 1 and 0, or only returns 0 and -1, it will not be able to sort reliably because anti-symmetry is broken. A comparator that always returns 0 will cause the array to not be changed at all, but is reliable nonetheless.

Reply:
Ah! good catch, I had this refactored to the same but I did not push it. Since any value < 0 will result in: `[a,b]` or the original as the sort result this is rather elegant.